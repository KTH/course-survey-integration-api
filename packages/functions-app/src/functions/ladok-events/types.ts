export type TLadokEventUserProperties = {
  content_HYPHEN_Type: 'application/json'
  ladok3AtomEntryId: string // 'b13bb75e-902e-11ee-81dd-dd57e0af499b'
  ladok3AtomEntryIndexInFeed: number // 0
  ladok3AtomEntryUpdated: string // '2023-12-01T10:47:50.358'
  ladok3AtomFeed: string // 'https://api.ladok.se:443/uppfoljning/feed/317137'
  ladok3EventId: string // 'b0c3c76e-902e-11ee-8455-d76a329e44d6'
  ladok3EventType: string // 'se.ladok.schemas.resultat.AnmalanPaAktivitetstillfalleEvent'
  ladok3IsLastFeed: boolean
  ladok3MessageSequenceNumber: number // 128310
  ladok3MessageType: 'ladok3Event'
  ladok3Username?: string // '[username]@kth.se'
  traceparent: string // '00-2981b58f9654ef80b29f4028d80565f0-2b59598bb4ce2c6b-01'
}

// - Utbildningsinformation:KurstillfalleEvent
// se.ladok.schemas.utbildningsinformation.KurstillfalleEvent
export type TKurstillfalle = {
}

// - Utbildningsinformation:KurstillfalleUppdateratEvent
// se.ladok.schemas.utbildningsinformation.KurstillfalleUppdateratEvent
export type TKurstillfalleUppdaterat = {
}

// - Utbildningsinformation:KurstillfalleTillStatusEvent
// se.ladok.schemas.utbildningsinformation.KurstillfalleTillStatusEvent
export type TKurstillfalleTillStatus = {
}

// - Studiedeltagande:RegistreringEvent
// se.ladok.schemas.studiedeltagande.RegistreringEvent
export type TRegistrering = {
}