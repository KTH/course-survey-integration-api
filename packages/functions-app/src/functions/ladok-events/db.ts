import { MongoClient } from "mongodb";

const { MONGODB_CONNECTION_STRING = "mongodb://localhost:27027/ladok3" } = process.env;
const DB_NAME = "ladok3";
const DB_QUEUE_NAME = "Ladok3FeedEvent";

const databaseClient = new MongoClient(MONGODB_CONNECTION_STRING, {
  maxPoolSize: 5,
  minPoolSize: 1,
  connectTimeoutMS: 1000,
});

let _dbOpenCounter = 0;
async function getCol() {
  _dbOpenCounter++;
  const database = databaseClient.db(DB_NAME);
  return database.collection(DB_QUEUE_NAME);
}

async function closeDb() {
  _dbOpenCounter--;
  if (_dbOpenCounter === 0) {
    // await databaseClient.close();
  }
}

export async function storeFeedEvent(message: any, userProps: any) {
  const event = {
    ladok3EventType: userProps.ladok3EventType,
    ladok3Username: userProps.ladok3Username,
    message,
    userProps,
  };

  let res;
  try {
    const eventCol = await getCol();
    res = await eventCol.insertOne({
      ...event,
    });
  } catch (e) {
    console.error(e);
  } finally {
    await closeDb();
  }

  if (res?.acknowledged) return;

  // TODO: Should we check reason?
  throw new Error("Could not insert entry into queue");
}