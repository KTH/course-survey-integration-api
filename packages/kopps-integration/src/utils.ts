import { InvalidLadokUidError, SyllabusNotFoundError } from "./errors";
import type {
  KoppsCourseDetailedInformation,
  KoppsCourseRoundInfo,
  KoppsSyllabus,
} from "./types";

const PERIODS = ["P0", "P1", "P2", "P3", "P4", "P5"] as const;
type Periods = (typeof PERIODS)[number];

/**
 * Return the periods in a round object. Periods are sorted from oldest to
 * latest by the term where the period belongs.
 */
export function getPeriods(roundObject: KoppsCourseRoundInfo) {
  const periods: Periods[] = [];
  const sortedTerms = roundObject.round.courseRoundTerms.sort(
    (t1, t2) => t1.term.term - t2.term.term,
  );

  for (const c of sortedTerms) {
    // Add the number to the "periods" array if the number of credits is higher than 0
    for (const p of PERIODS) {
      const numberOfCredits = c[`credits${p}`];
      if (numberOfCredits && numberOfCredits > 0) {
        periods.push(p);
      }
    }
  }

  return periods;
}

/** Return the course round information given a Ladok UID */
export function getCourseRoundInformation(
  detailedInformation: KoppsCourseDetailedInformation,
  ladokUid: string,
): KoppsCourseRoundInfo {
  const roundInformation = detailedInformation.roundInfos.find(
    (r) => r.round.ladokUID === ladokUid,
  )!;

  if (!roundInformation) {
    throw new InvalidLadokUidError(ladokUid);
  }

  return roundInformation;
}

/** Return the most recent syllabus that is valid in the given term */
export function getSyllabus(
  allSyllabus: KoppsSyllabus[],
  term: number,
): KoppsSyllabus {
  // Filter-out the invalid ones
  const validSyllabus = allSyllabus.filter((s) => s.validFromTerm.term <= term);

  if (validSyllabus.length === 0) {
    throw new SyllabusNotFoundError(term);
  }

  // Take the latest
  return validSyllabus.sort(
    (s1, s2) => s2.validFromTerm.term - s1.validFromTerm.term,
  )[0];
}
