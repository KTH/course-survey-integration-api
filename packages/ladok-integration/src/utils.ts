import type { z } from "zod";
import { LadokOrganisation } from "./types";
import assert from "node:assert";

type LadokOrganisation = z.infer<typeof LadokOrganisation>;

/**
 * This information is obtained from Ladok
 * But hardcoded for eficiency reasons
 */
export function getGradingScheme(id: number) {
  // AF
  if (id === 131657) {
    return {
      code: "AF",
      grades: [
        { validFinalGrade: true, code: "A" },
        { validFinalGrade: true, code: "B" },
        { validFinalGrade: true, code: "C" },
        { validFinalGrade: true, code: "D" },
        { validFinalGrade: true, code: "E" },
        { validFinalGrade: false, code: "FX" },
        { validFinalGrade: false, code: "F" },
      ],
    };
  }

  // PF
  if (id === 131656) {
    return {
      code: "PF",
      grades: [
        { validFinalGrade: true, code: "P" },
        { validFinalGrade: false, code: "F" },
      ],
    };
  }

  throw new Error(`Unknown grading scheme id [${id}]`);
}

export function parseOrganisation(organisation: LadokOrganisation) {
  // Both "Benamning.sv" and "Benamning.en" are formatted as
  // "SCI/Matematik" where "SCI" is the School code and should be the same
  // in both languages
  const [codeSv, departmentSv] = organisation.Benamning.sv.split("/");
  const [codeEn, departmentEn] = organisation.Benamning.en.split("/");

  // Verifying the length of the arrays
  if (!departmentEn) {
    throw new Error(
      `Incompatible format. Benamning.en must be "School code/Institution name". Obtained: [${organisation.Benamning.en}]`,
    );
  }

  if (!departmentSv) {
    throw new Error(
      `Incompatible format. Benamning.sv must be "School code/Institution name". Obtained: [${organisation.Benamning.sv}]`,
    );
  }

  if (codeSv !== codeEn) {
    throw new Error(
      `School code in Swedish [${codeSv}] does not match with code in English [${codeEn}]`,
    );
  }

  return {
    school: {
      code: codeSv,
    },
    department: {
      code: organisation.Organisationskod,
      name: {
        en: departmentEn,
        sv: departmentSv,
      },
    },
  };
}

/**
 * Get the year and term from a date
 * @param {string} date  Date formatted as `YYYY-MM-DD`
 * @return a string "XXXXY" where XXXX is a year and Y is the term (1 for spring, 2 for autumn)
 */
export function getTermFromDate(date: string) {
  const REGEX = /(\d\d\d\d)\-(\d\d)\-(\d\d)/;
  const m = date.match(REGEX);

  assert(m !== null, "Error!!1");

  const year = m[1];
  const month = parseInt(m[2], 10);

  if (month <= 6) {
    return `${year}1`;
  } else {
    return `${year}2`;
  }
}

/** Get the difference between two terms */
export function diffTerms(t1: string, t2: string) {
  // 1. Number of years between the periods
  const yDiff = parseInt(t1.slice(0, 4), 10) - parseInt(t2.slice(0, 4), 10);

  // 2. If they are not the same period, add or substract 1
  const tDiff = parseInt(t1.slice(-1), 10) - parseInt(t2.slice(-1), 10);

  return yDiff * 2 + tDiff;
}

/**
 * Given the reference of a "Studiestruktur" and a tree with "Studiestruktur",
 * find the leaf in the structure with the reference
 *
 * Returns an array of the path from the root to the found leaf
 */
export function findStudiestruktur<T extends { Referens: string; Barn: T[] }>(
  ref: string,
  struktur: T[],
): Omit<T, "Barn">[] {
  for (const leaf of struktur) {
    const { Barn, ...other } = leaf;

    if (leaf.Referens === ref) {
      return [other];
    }

    if (leaf.Barn.length > 0) {
      const subStructure = findStudiestruktur(ref, leaf.Barn);

      if (subStructure.length > 0) {
        return [other, ...subStructure];
      }
    }
  }

  return [];
}
