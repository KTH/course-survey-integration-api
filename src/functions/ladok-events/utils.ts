import * as crypto from "crypto";
import { TUgUser, TUgSchool } from "ug-integration";
import { TCourseUser, TOrgEntity } from "../interface";

export function _convert(ugUser: TUgUser): TCourseUser {
  const email = ugUser.email;
  const userName = email.split("@")[0];
  const kthUserId = ugUser.kthid;
  const fullName = [ugUser.givenName, ugUser.surname].join(" ").trim();

  return {
    userName,
    kthUserId,
    email,
    fullName,
  };
}

export function convertUgToCourseUser(
  ugUser: TUgUser | undefined,
): TCourseUser | undefined {
  if (ugUser === undefined) return undefined;
  return _convert(ugUser);
}

export function convertUgToCourseUserArr(
  ugUser: (TUgUser | undefined)[] | undefined,
): TCourseUser[] {
  if (Array.isArray(ugUser)) {
    return (ugUser.filter((o) => o) as TUgUser[]).map(_convert);
  }
  return [];
}

export function convertUgSchoolToOrgEntity(
  ugSchool: TUgSchool | undefined,
  schoolCode: string,
  lang: "en" | "sv",
): TOrgEntity {
  return {
    displayName: ugSchool?.description[lang] ?? "",
    displayCode: schoolCode.toUpperCase(),
    kthId: ugSchool?.kthid ?? "",
  };
}

export function convertLadokModuleToCourseModule(
  ladokModule: any,
  lang: string,
): any {
  const gradingDistribution = ladokModule.gradingScheme?.grades?.reduce(
    (val: any, curr: any) => {
      return {
        ...val,
        [curr.code]: -1,
      };
    },
    {},
  );

  return {
    _reportedResults: {},
    code: ladokModule.code,
    name: ladokModule.name?.[lang],
    credits: ladokModule.credits,
    gradingScheme: Object.keys(gradingDistribution ?? {}),
    nrofReportedResults: 0, // TODO: Calculated field
    gradingDistribution, // TODO: Calculated field
  };
}

export async function hashStudentId(studentId: string): Promise<string> {
  const hash = crypto.createHash("sha256");

  return new Promise((resolve, reject) => {
    hash.on("error", (err) => {
      reject(err);
    });

    hash.on("readable", () => {
      // Only one element is going to be produced by the
      // hash stream.
      const data = hash.read();
      if (data) {
        const hash = data.toString("hex");
        resolve(hash);
        // Prints:
        //   6a2da20943931e9834fc12cfe5bb47bbd9ae43489a30726962b576f4e3993e50
      }
    });

    hash.write(studentId);

    hash.end();
  });
}
