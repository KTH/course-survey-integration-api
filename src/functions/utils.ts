import { HttpRequest, InvocationContext, HttpResponseInit } from "@azure/functions";
import { MongoClient, ObjectId } from 'mongodb';
import { TLadokEventUserProperties } from "./ladok-events/types";

const IS_PROD = process.env.NODE_ENV === "production";
const { COSMOSDB_CONNECTION_STRING } = process.env;
if (IS_PROD && !COSMOSDB_CONNECTION_STRING) throw new Error("Missing env var COSMOS_DB_CONNECTION_STRING");

type ApiHttpHandler = (req: HttpRequest, context: InvocationContext, db: Database) => Promise<HttpResponseInit>;

/**
 * This wrapper functions makes sure that the entire process doesn't crash on errors
 * by wrapping it in a try/catch. See note in code.
 * @param handler
 * @returns
 */
export function API(handler: ApiHttpHandler) {
  return async function (req: HttpRequest, ctx: InvocationContext) {
    try {
      ctx.log(`Http function processed request for url "${req.url}"`);
      return await handler(req, ctx, new Database());
    } catch (err) {
      ctx.error(err);
      // This rethrown exception will only fail the individual invocation, instead of crashing the whole process
      // https://learn.microsoft.com/en-us/azure/azure-functions/functions-reference-node?pivots=nodejs-model-v4&tabs=typescript%2Cwindows%2Cazure-cli#use-async-and-await
      throw err;
    }
  };
}

type ServiceBusTopicHandler<T = unknown> = (message: T, context: InvocationContext, db: Database) => Promise<void>;
/**
 * This wrapper functions makes sure that the entire process doesn't crash on errors
 * by wrapping it in a try/catch. See note in code.
 * @param handler
 * @returns
 */
export function ServiceBus<T>(handler: ServiceBusTopicHandler<T>) {
  return async function (msg: unknown, ctx: InvocationContext) {
    try {
      return await handler(<T>msg, ctx, new Database());
    } catch (err) {
      ctx.error(err);
      // This rethrown exception will only fail the individual invocation, instead of crashing the whole process
      // https://learn.microsoft.com/en-us/azure/azure-functions/functions-reference-node?pivots=nodejs-model-v4&tabs=typescript%2Cwindows%2Cazure-cli#use-async-and-await
      throw err;
    }
  };
}

// Runtime check to make sure that we don't process events that are not meant for us
export function isValidEvent(eventName: string, userProperties: unknown): boolean {
  return (<TLadokEventUserProperties>userProperties)?.ladok3EventType === eventName;
}

export type DbCollectionName = "StudentParticipation" | "CourseRound" | "Module" | "Program" | "ReportedResult";

export class Database {
  _client: MongoClient | undefined;

  async connect(): Promise<void> {
    if (this._client) return;
    const mongoClient = new MongoClient(COSMOSDB_CONNECTION_STRING!);
    this._client = await mongoClient.connect();
  }
  async close(): Promise<void> {
    if (this._client) await this._client.close();
    this._client = undefined;
  }

  async fetchById<T>(id: string, collectionName: DbCollectionName): Promise<any> {
    await this.connect();
    const collection = this._client!.db().collection(collectionName);
    const doc = await collection.findOne({ id });
    return doc as T;
  }

  async queryByProperty<T>(propName: string, value: string | object, collectionName: DbCollectionName): Promise<any[]> {
    await this.connect();
    const collection = this._client!.db().collection(collectionName);
    const docs = await collection.find({ [propName]: value }).toArray();
    return docs as T[];
  }

  async query<T>(query: { [key: string]: any }, collectionName: DbCollectionName): Promise<any[]> {
    await this.connect();
    const collection = this._client!.db().collection(collectionName);
    const docs = await collection.find(query).toArray();
    return docs as T[];
  }

  async countByPropertyQuery(propName: string, value: string, collectionName: DbCollectionName): Promise<number> {
    await this.connect();
    const collection = this._client!.db().collection(collectionName);
    const count = await collection.countDocuments({ [propName]: value });
    return count;
  }

  async insert<T extends TBaseEntity>(doc: T, collectionName: DbCollectionName): Promise<void> {
    await this.connect();
    const collection = this._client!.db().collection(collectionName);
    await collection.insertOne({ ...doc });
  }

  async update<T extends TBaseEntity>(id: string, partial: Partial<T>, collectionName: DbCollectionName): Promise<void> {
    await this.connect();
    const collection = this._client!.db().collection(collectionName);
    await collection.updateOne({ id }, partial);
  }
}

type TBaseEntity = {
  id: string;
};
