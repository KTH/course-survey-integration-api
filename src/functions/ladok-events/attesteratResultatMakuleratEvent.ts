import { InvocationContext } from "@azure/functions";
import { TLadokEventContext } from "./types";
import { Database, ServiceBus, isValidEvent } from "../utils";

export type TAttesteratResultatMakuleratEvent = {
  Beslut: {
    BeslutUID: string; // "34b1ff17-603e-11e9-9dcc-b1e66e1540b0",
    Beslutsdatum: string; // "2024-01-18",
    Beslutsfattare: string; // "Emil Stenberg (IT)",
    BeslutsfattareUID: string; // "34b1ff17-603e-11e9-9dcc-b1e66e1540b0"
  };
  GiltigSomSlutbetyg: boolean; // true,
  KursUID: string; // "a4565647-b3b8-11ee-bf6a-e2af0a9345af",
  PrestationsPoang: number; // 5,
  ResultatFinns: boolean; // false,
  ResultatUID: string; // "31d936dc-b53f-11ee-988f-6acd08c746d6",
  StudentUID: string; // "ea667761-2c55-11ee-998a-6a4f18c04b6e",
  UtbildningensOmfattning: number; // 5,
  UtbildningsgrundtypID: number; // 1,
  UtbildningsinstansUID: string; // "715b159a-b536-11ee-aef4-bd1cad394d29",
  HandelseUID: string; // "5c6e3eab-b5de-11ee-988f-6acd08c746d6",
  EventContext: TLadokEventContext;
};

export async function handler(
  message: TAttesteratResultatMakuleratEvent,
  context: InvocationContext,
  db: Database,
): Promise<void> {
  if (
    !isValidEvent(
      "AttesteratResultatMakuleratEvent",
      context?.triggerMetadata?.userProperties,
    )
  )
    return;

  context.log(`AttesteratResultatMakuleratEvent: `);
}

export default {
  handler: ServiceBus<TAttesteratResultatMakuleratEvent>(handler),
  // input binding doesn't support cosmos document store yet
  // extraInputs: [cosmosInput],
  // extraOutputs: [cosmosOutput],
};
