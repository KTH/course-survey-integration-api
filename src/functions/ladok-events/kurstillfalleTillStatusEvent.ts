import { InvocationContext } from "@azure/functions";
import { TLadokEventContext, TLadokAttributvarde } from "./types";
import { Database, ServiceBus, isValidEvent } from "../utils";

export type TKurstillfalleTillStatusEvent = {
  HandelseUID: string, // "7c2e425f-9507-11ee-a0ce-a9a57d284dbd",
  EventContext: TLadokEventContext,
  OrganisationUID: string, // "e842afc9-73d6-11e8-8c58-f9aa7f7e4fb6",
  Status: number, // 2,
  UtbildningUID: string, //  "e51b9586-9501-11ee-a0ce-a9a57d284dbd",
  Attributvarde: TLadokAttributvarde[],
  /*
  {
    "Grupp": 0,
    "Namn": "utbildning.attribut.undervisningstid",
    "Varde": "101052",
    "Uid": "3ea87095-9507-11ee-a0ce-a9a57d284dbd"
  },
  {
    "Grupp": 0,
    "Namn": "utbildning.attribut.finansieringsform",
    "Varde": "102085",
    "Uid": "3ea87096-9507-11ee-a0ce-a9a57d284dbd"
  },
  {
    "Grupp": 0,
    "Namn": "utbildning.attribut.studieavgiftsbelagd",
    "Varde": "false",
    "Uid": "3ea87097-9507-11ee-a0ce-a9a57d284dbd"
  }
  */
  UtbildningsinstansUID: string, // "e51b9585-9501-11ee-a0ce-a9a57d284dbd",
  UtbildningstypID: number, // 52,
  Avgiftsbelagt: boolean, // false,
  StartperiodID: number, // 151878,
  StudielokaliseringID: number, // 135195,
  StudietaktID: number, // 1,
  UndervisningsformID: number, // 1,
  UndervisningstidID: number, // 101052,
  UtbildningUtbildningstypID: number, // 22,
  UtbildningstillfalleUID: string, // "41717c91-4028-11ee-bf53-2115569549a8",
  Utbildningstillfalleskod: string, // "12349",
  Kurstillfallesattribut: {
    Finansieringsform: number, // 102085
  },
  Kurstillfallesperioder: {
    ForstaRegistreringsdatum: string, // "2023-08-28",
    ForstaUndervisningsdatum: string, // "2023-08-28",
    Omfattningsvarde: number, // 10,
    SistaRegistreringsdatum: string, // "2024-01-15",
    SistaUndervisningsdatum: string, // "2024-01-15",
    UtbildningsomradePerOrganisation: {
      OrganisationUID: string, // "e842afc9-73d6-11e8-8c58-f9aa7f7e4fb6",
      Procent: number, // 100,
      UtbildningsomradeID: number, // 2,
      Uid: string, // "3ea8709a-9507-11ee-a0ce-a9a57d284dbd"
    }[],
    Uid: string, // "3ea87099-9507-11ee-a0ce-a9a57d284dbd"
  }[]
}

export async function handler(message: TKurstillfalleTillStatusEvent, context: InvocationContext, db: Database): Promise<void> {
  if (!isValidEvent("se.ladok.schemas.utbildningsinformation.KurstillfalleTillStatusEvent", context?.triggerMetadata?.userProperties)) return;

  const utbildningstillfalleUid = message.UtbildningstillfalleUID;
  const status = message.Status;
  context.log(`KurstillfalleTillStatusEvent: ${utbildningstillfalleUid} ${status}`);

  try {
    const courseRound = await db.fetchById(utbildningstillfalleUid, "CourseRound");
    // 1. Fetch CourseRound from DB
    // 2. Update status
    // 3. Persist in DB
  
    await db.update(courseRound.id!, { status }, "CourseRound");
  } finally {
    await db.close();
  }
}

export default {
  handler: ServiceBus<TKurstillfalleTillStatusEvent>(handler),
  extraInputs: undefined,
  extraOutputs: undefined,
}
