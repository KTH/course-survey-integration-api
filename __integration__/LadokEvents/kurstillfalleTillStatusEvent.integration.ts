import event from "../../__fixtures__/events/01_utbildningsinformation.KurstillfalleTillStatusEvent_1.json";
import { handler } from "../../src/functions/ladok-events/kurstillfalleTillStatusEvent";
import { MockContext } from "../../__tests__/LadokEvents/utils/mockContext";
import { MockDatabase } from "../../__tests__/LadokEvents/utils/mockDatabase";

const DB_MOCK_INPUT = {
  "CourseRound": {
    "id": "41717c91-4028-11ee-bf53-2115569549a8",
  }
}

describe("ModulTillStatusEvent", () => {

  test("can be executed", async () => {
    const mockDb = new MockDatabase(DB_MOCK_INPUT);
    const mockContext = new MockContext(event.userProps);
    await handler(event.message, mockContext, mockDb);
  });

  test("writes correct data to db", async () => {
    const mockDb = new MockDatabase(DB_MOCK_INPUT);
    const mockContext = new MockContext(event.userProps);
    await handler(event.message, mockContext, mockDb);

    expect(mockDb._result).toMatchSnapshot();
  });
});
