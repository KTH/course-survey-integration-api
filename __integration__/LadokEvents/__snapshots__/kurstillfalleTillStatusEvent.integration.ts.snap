// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModulTillStatusEvent writes correct data to db 1`] = `
{
  "CourseRound": {
    "id": "41717c91-4028-11ee-bf53-2115569549a8",
    "status": 2,
  },
}
`;
